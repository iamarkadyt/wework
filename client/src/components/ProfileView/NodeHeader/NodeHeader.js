import React from 'react'
import Field from '../../Field/Field'
import { string, bool, func } from 'prop-types'

/**
 * Header for experience & education nodes.
 * Outputs {company} name with {timespan} or {delButton} 
 */
const NodeHeader = ({
    title,
    from,
    to,
    current,
    showDelButton,
    onDelBtnClick
}) => {
    // Month ##, ####
    const dateFormatOptions = { year: 'numeric', month: 'long', day: 'numeric' }

    from = new Date(from).toLocaleDateString('en-US', dateFormatOptions)
    to = current
        ? 'Current'
        : new Date(to).toLocaleDateString('en-US', dateFormatOptions)
    return (
        <div style={{ display: 'flex', justifyContent: 'space-between' }}>
            <h3>{title}</h3>
            {showDelButton
                ? <Field
                    type="linkButton"
                    label="Delete"
                    containerStyle={{ margin: 0, width: 'auto' }}
                    onClick={onDelBtnClick} />
                : <span>{from} â€” {to}</span>}
        </div>
    )
}

NodeHeader.propTypes = {
  title: string.isRequired,
  from: string.isRequired,
  to: string,
  current: bool,
  showDelButton: bool.isRequired,
  onDelBtnClick: func.isRequired
}

export default NodeHeader
