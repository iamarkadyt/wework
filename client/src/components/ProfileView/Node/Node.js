import React from 'react'
import NodeHeader from '../NodeHeader/NodeHeader'
import { arrayOf, string, bool, func } from 'prop-types'

const node = ({
  _id,
  headers,
  from,
  to,
  current,
  description,
  onDelBtnClick,
  isDeleting
}) => {
  return (
    <div className="node">
      <NodeHeader
        title={headers[0]}
        from={from}
        to={to}
        current={current}
        showDelButton={isDeleting}
        onDelBtnClick={() => onDelBtnClick(_id)} />
      {headers.map((item, idx) => {
        // skip the first header as it's shown in NodeHeader
        if (idx === 0) 
          return undefined

        return (
          <p key={idx}>
            {headers[idx]}
          </p>
        )
      })}
      {description
        ? <p><b>Description:</b> {description}</p>
        : null}
    </div>
  )
}

node.propTypes = {
  _id: string.isRequired,
  headers: arrayOf(string).isRequired,
  from: string.isRequired,
  to: string,
  current: bool,
  description: string,
  onDelBtnClick: func.isRequired,
  isDeleting: bool.isRequired
}

export default node
