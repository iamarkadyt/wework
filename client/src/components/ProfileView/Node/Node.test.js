import React from 'react'
import Node from './Node'
import NodeHeader from '../NodeHeader/NodeHeader'
import { shallow } from 'enzyme'
import cloneDeep from 'lodash.clonedeep'

describe('Node', () => {
  let mountedComponent, props

  const getMockProps = () => {
    return cloneDeep({
      _id: 'someid_1298',
      headers: [ 'header0', 'header1' ],
      from: 'from',
      to: 'to',
      onDelBtnClick: jest.fn(),
      isDeleting: true,
      description: 'desc'
    })
  }

  const comp = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Node {...props} />)
    } 
    return mountedComponent
  }

  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('props', () => {
    describe('effects', () => {
      describe('description', () => {
        it('if undefined, comp doesn\'t render description paragraph', () => {
          delete props.description
          expect(comp().find('p b').exists()).toBe(false)
        })

        it('if defined, comp renders description paragraph', () => {
          props.description = 'some text'
          expect(comp().find('p b').exists()).toBe(true)
        })
      })
    })
  })

  describe('functions', () => {
    it('calls onDelBtnClick', () => {
      comp().find(NodeHeader).simulate('delBtnClick')
      expect(props.onDelBtnClick).toHaveBeenCalled()
    })

    it('calls onDelBtnClick with correct args', () => {
      comp().find(NodeHeader).simulate('delBtnClick')
      expect(props.onDelBtnClick).toHaveBeenCalledWith(props._id)
    })
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})

