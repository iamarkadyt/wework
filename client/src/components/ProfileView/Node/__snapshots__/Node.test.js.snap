// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node matches snapshot 1`] = `
<div
  className="node"
>
  <NodeHeader
    from="from"
    onDelBtnClick={[Function]}
    showDelButton={true}
    title="header0"
    to="to"
  />
  <p
    key="1"
  >
    header1
  </p>
  <p>
    <b>
      Description:
    </b>
     
    desc
  </p>
</div>
`;
